### PR TITLE
Step VR 1

### PR DESCRIPTION
**Pull request by JoshuaMitchell**

### CODE REVIEW

File: app/main/VR/VR1_citizenship.py [hunk 1]
```diff
@@ -1,16 +1,29 @@
 from app.main import main
-from flask import g, url_for, render_template, redirect, request
+from flask import g, url_for, render_template, redirect, request, session as sess
```
Comment (karpet) [inline app/main/VR/VR1_citizenship.py]: I know the word `session` is starting to get overloaded in this app. Could we agree on a naming convention so it is super clear what we're looking at? I suggest `http_session` `db_session` etc for clarity.

File: app/main/tests/test_step_vr_1.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from app.models import *
+from flask import g
+def create_registrant(session):
```
Comment (karpet) [inline app/main/tests/test_step_vr_1.py]: `db_session` instead of `session`?

File: app/main/tests/test_step_vr_1.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        },
+        county="Douglas",
+        reg_lookup_complete = True,
+    )
+    session.add(registrant)
+    session.commit()
+    return registrant
+
+
+def test_citizenship_not_checked_does_not_return_redirect(app, session, client):
```
Comment (karpet) [inline app/main/tests/test_step_vr_1.py]: `db_session`?

File: app/main/tests/test_step_vr_1.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    """
+    An existing user tries to update their record, but does not select the citizen checkbox
+    """
+    registrant = create_registrant(session)
+    with client.session_transaction() as sess:
+        sess['session_id'] = str(registrant.session_id)
```
Comment (karpet) [inline app/main/tests/test_step_vr_1.py]: `http_session`?

File: app/main/tests/test_step_vr_1.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    form_payload = {}
+    response = client.post('/vr/citizenship', data=form_payload, follow_redirects=False)
+    assert response.status_code != 302
+
+def test_citizenship_checked_returns_redirect(app,session,client):
```
Comment (karpet) [inline app/main/tests/test_step_vr_1.py]: `db_session`?

File: app/main/tests/test_step_vr_1.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    registrant = create_registrant(session)
+    with client.session_transaction() as sess:
```
Comment (karpet) [inline app/main/tests/test_step_vr_1.py]: `http_session` instead of `sess`?

File: app/services/session_manager.py [hunk 7]
```diff
@@ -20,14 +20,16 @@ def _init_next_step(self):
         If the current step has a next step set, initialize the next step class and save it to self.
         """
         if self.current_step.next_step:
-            self.next_step = globals()[self.current_step.next_step]
+            object = globals()[self.current_step.next_step]
```
Comment (karpet) [inline app/services/session_manager.py]: isn't everything an object? could we name this `next_step` instead? or `next_step_instance`?

File: app/services/session_manager.py [hunk 8]
```diff
@@ -47,9 +48,10 @@ def get_redirect_url(self):
 
         # For the previous step iterate all of the requirements.
         # If the requirement is not fulfilled return the previous step url
-        for req in self.prev_step:
+        for req in self.prev_step.all_requirements():
             # if a requirement is missing return the endpoint for the previous step
             if not self.registrant.has_value_for_req(req):
+                print(req)
```
Comment (karpet) [inline app/services/session_manager.py]: debugging line?

Comment (karpet) [reply app/services/session_manager.py]: do we have unit tests just for `SessionManager` class? That might be worthwhile at this point.

File: app/main/tests/test_step_vr_1.py (hunk 2 above)
Comment (JoshuaMitchell) [reply app/main/tests/test_step_vr_1.py]: Probably not a bad idea.  Maybe that's a fix pull request to rename all the sessions db vs http

File: app/services/session_manager.py (hunk 7 above)
Comment (JoshuaMitchell) [reply app/services/session_manager.py]: done

File: app/services/session_manager.py (hunk 8 above)
Comment (JoshuaMitchell) [reply app/services/session_manager.py]: that was a debugging line.  it is removed now.  Don't currently have unittests for session manager.  might deserve its own pr.

**Comment by JoshuaMitchell** [comment]
@karpet if your're good with SessionManager unittests being its own pr and renaming ambiguous session variables as its own PR i think this is good to squash and merge in.

**Pull request merged**